packages/core-plugin-api/src/extensions/extensions.tsx

```diff
@@ -51,8 +51,27 @@ export type ComponentLoader<T> =
 export function createRoutableExtension<
   T extends (props: any) => JSX.Element | null,
 >(options: {
+  /**
+   * A loader for the component that is rendered by this extension.
```

Comment:
Could be nice to here or in the function docstring to have an `@example` so the use of `lazy` and the name convention is more readily apparent